lighthouse-core/lib/network-request.js

```diff
@@ -246,6 +247,18 @@ module.exports = class NetworkRequest {
 
     this.responseReceivedTime = timestamp;
 
+    this.responseHeadersText = response.headersText || '';
+    this.responseHeaders = NetworkRequest._headersDictToHeadersArray(response.headers);
+
+    // The total length of the encoded data is spread out among multiple events. The sum of the
+    // values in onResponseReceived and all the onDataReceived events will equal the value
+    // seen on the onLoadingFinished event. As we process onResonseReceived and onDataReceived
+    // we accumulate the total encodedDataLength. When we process onLoadingFinished, we override
+    // the accumulated total. We do this so that if the request is aborted or fails, we still get
+    // a value via the accumulation.
+    // In Lightrider, we do not have true value for encodedDataLength, and we get the actual size
```

Comment:
> we do no accumulation.

Looks like we do, as of now.. We just override the transferSize in loadingFinished

Thank you VERY MUCH for writing all this shit out. These comments are so Good. 

here's a very small rewrite of this second para for clarity:

> In Lightrider, due to instrumentation limitations, our values for encodedDataLength are bogus and not valid. However the resource's true `encodedDataLength`/`transferSize` is shared via a  special response header, `X-TotalFetchedSize`. In this situation, we read this value from responseReceived, use it for the `transferSize` and ignore the encodedDataLength values in both dataReceived and loadingFinished.

Reply:
oops thanks for the revisions :heart: